velox/expression/tests/CastExprTest.cpp

```diff
@@ -943,6 +1063,10 @@ TEST_F(CastExprTest, toString) {
   ASSERT_EQ("cast((a) as ARRAY<VARCHAR>)", exprSet.exprs()[1]->toString());
 }
 
+TEST_F(CastExprTest, floatingToDecimal) {
+  testFloatingToDecimalCasts<double>();
```

Comment:
Can we test for float type as well?

Reply:
Currently, no, we can't. The float number has a very low precision. For example:

```cpp
float num = 99999.99f;
std::cout << std::fixed << num << std::endl;
```

The output will be `99999.992188`.

In presto, the value about `select cast(99999.99 as decimal(10,4));` is `99999.9900`, but in this unit test, the value is `99999.9921`.

So in fact, we haven't support float yet.